app/controllers/holds_controller.rb

```diff
@@ -35,6 +37,56 @@ def new
     @bib = Bib.new(parsed_body, holdable_locations)
   end
 
+  # Handles placing holds
+  #
+  # POST /holds
+  def create
+    results = barcodes.each_with_object(success: [], error: []) do |barcode, status|
+      hold_args = { pickup_library: params['pickup_library'], pickup_by_date: params['pickup_by_date'] }
+      response = symphony_client.place_hold(patron,
+                                            current_user.session_token,
+                                            barcode,
+                                            hold_args)
+      case response.status
+      when 200
+        status[:success] << { barcode: barcode,
+                              hold_key: JSON.parse(response.body).dig('holdRecord', 'key') }
+      else
+        Rails.logger.error("Place Hold for #{barcode} by #{patron.barcode}: #{response.body}")
+        status[:error] << { barcode: barcode,
+                            error_message: JSON.parse(response.body).dig('messageList')[0].dig('message') }
+      end
+    end
+
+    session[:place_hold_results] = results
+    session[:place_hold_catkey] = params['catkey']
+
+    redirect_to result_path
+  end
+
+  # Handles place hold response
+  #
+  # GET /holds/result
+  def result
+    (redirect_to holds_path) && return if session[:place_hold_results].blank?
```

Comment:
```suggestion
    return redirect_to holds_path if session[:place_hold_results].blank?
```

I think this is a bit easier to read, the `&&` kinda through me tbh. Guess we could just `raise` here too, or, simply not validate at all and let a custom error page handle the error this will throw. It should be a pretty rare case that someone would go here directly. (like never?)

Reply:
Happens when someone goes to /holds/result directly or if try to refresh the result page they are on, yes probably rare.